lib/records/unit-tests/RecProtoTagsTest.cpp

```diff
@@ -0,0 +1,112 @@
+/** @file
+  Test file for basic_string_view class
+  @section license License
+  Licensed to the Apache Software Foundation (ASF) under one
+  or more contributor license agreements.  See the NOTICE file
+  distributed with this work for additional information
+  regarding copyright ownership.  The ASF licenses this file
+  to you under the Apache License, Version 2.0 (the
+  "License"); you may not use this file except in compliance
+  with the License.  You may obtain a copy of the License at
+      http://www.apache.org/licenses/LICENSE-2.0
+  Unless required by applicable law or agreed to in writing, software
+  distributed under the License is distributed on an "AS IS" BASIS,
+  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+  See the License for the specific language governing permissions and
+  limitations under the License.
+ */
+
+#include "catch.hpp"
+#include "records/RecProtoTags.h"
+#include "ink_memory_stubs.h"
+
+#ifndef _DEBUG
+#define _DEBUG
+#endif
+
+#include <string>
+#include "ts/MemView.h"
+#include "records/RecProtoTags.h"
+
+const ts::StringView IP_PROTO_TAG_IPV4("ipv4", ts::StringView::literal);
```

Comment:
Why declare these as StringView if you are going to only use the ptr()?  If you use const char* you should be able to get rid of including ts/MemView.h and possibly stubbing out ats_malloc|free.

Reply:
Sure I can try that.

Reply:
They are used directly as StringView instances in `ink_inet.cc` and other protocol stack methods.

Reply:
I agree in that we should be using the internal code. I believe the issue Derek is trying to address with the issue with the ball of wax in our depends chains. in this case we have something like ( simplified):

```
records:
            libts
            mgmt.api
            eventsystem
                         records
              proxy
                         records
```

This is why I was going to originally make one large unit test as our code currently is going to require most of the test code to link with individual objects ( which is annoying hard in automake)... or we link in most of the other .a file in a certain order

Reply:
Fixed by linking to `libtsutil` to avoid duplicating these.